sdk/storage/azure-storage-common/src/storage_credential.cpp

```diff
@@ -96,10 +96,18 @@ namespace Azure { namespace Storage { namespace _internal {
     std::string sas = getWithDefault(connectionStringMap, "SharedAccessSignature");
```

Comment:
Question: Rather than building a map from the connection string and then doing this look up to build `connectionStringParts`, why not build the parts directly at this point?
https://github.com/Azure/azure-sdk-for-cpp/blob/c455f5d6d65eeaf24f103240b4ffeb88b5b12aca/sdk/storage/azure-storage-common/src/storage_credential.cpp#L12-L40

Reply:
Parsing connection string is kind of complicated. The code is long and not easy to understand, especially when mixed with other code logics.

So I prefer separating the code logics into two parts, first parsing connection string, then building the urls.